Source/UZKArchive.m

```diff
@@ -1973,6 +1999,11 @@ - (BOOL)performWriteAction:(int(^)(uLong *crc, NSError * __autoreleasing*innerEr
         UZKLogDebug("Making zip_fileinfo struct for date %{time_t}ld", lrint(fileDate.timeIntervalSince1970));
         zip_fileinfo zi = [UZKArchive zipFileInfoForDate:fileDate];
         
+        if (posixPermissions > 0) {
+            
+            zi.external_fa = (posixPermissions + 32768) << 16;
```

Comment:
Can you please define a constant for `32768` on the previous line to self-document its purpose? Also, just so I know, why do we need to add this constant and then bit shift?

Reply:
Since the permission value accessed by NSFileManager is octal, some offset conversion is performed.